File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.go
func twoSum(numbers []int, target int) []int {
	result := []int{0, 0}
	left, right := 0, len(numbers)-1
	for left < right {
		sum := numbers[left] + numbers[right]
		if sum == target {
			result[0] = left + 1
			result[1] = right + 1
			return result
		} else if sum > target {
			right--
		} else {
			left++
		}

	}
	return result
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        HashMap<Integer, Integer> hash = new HashMap<>();
        for(int i = 0; i < n; i++) {
            hash.put(nums[i], i);
        }
        System.out.println(hash);
        int[] result = new int[2];
        for (int i = 0; i < n; i++) {
            int difference = target - nums[i];
            if (hash.containsKey(difference) && hash.get(difference) != i) {
                result[0] = i;
                result[1] = hash.get(difference);
                return result;
            } else {
                continue;
            }
        }
        return result;
    }
}